airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -742,6 +745,7 @@ def create_dpath_flatten_fields(
             delete_origin_value=model.delete_origin_value
             if model.delete_origin_value is not None
             else False,
+            replace_record=model.replace_record if model.replace_record is not None else False,
```

Comment:
I think it could be just: `model.replace_record or False`; please test it before merging.